src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1842,6 +1843,26 @@ public static function preventsLazyLoading()
         return static::$modelsShouldPreventLazyLoading;
     }
 
+    /**
+     * Get the broadcast channel route definition that is associated with the given entity.
+     *
+     * @return string
+     */
+    public function broadcastChannelRoute()
+    {
+        return str_replace('\\', '.', get_class($this)).'.{'.$this->getKeyName().'}';
```

Comment:
Maybe we could default to using the class basename for the wildcard? This way we would allow using the implicit binding:

```php
Broadcast::channel(ModelName::class, function (User $user, ModelName $modelName) {
  return $user->belongsToTeam($modelName->team);
});
```

Or we could allow configuring the wildcard name as a tuple:

```php
Broadcast::channel([ModelName::class, 'myModel'], function (User $user, ModelName $myModel) {
  return $user->belongsToTeam($myModel->team);
});
```

Or getting more "advanced" and use reflection to find the variable name of the model type-hinted with the class passed to generate the channel, don't know.

With that being said, having it as the key name wouldn't be a problem though:

```php
Broadcast::channel(ModelName::class, fn ($user, $id) => 
  $user->belongsToTeam(Team::findOrFail($id))
);
```

Reply:
Ah good call - forgot about the implicit binding. I can update this to be the class basename (in camelCase).

Reply:
Updated.